network/src/connectivity_manager/mod.rs

```diff
@@ -205,6 +223,36 @@ impl DiscoveredPeer {
     pub fn is_eligible_to_be_dialed(&self) -> bool {
         self.is_eligible() && !self.addrs.is_empty()
     }
+
+    /// Updates the last time we tried to connect to this node
+    pub fn set_last_dial_time(&mut self, time: SystemTime) {
+        self.last_dial_time = time;
+    }
+
+    /// Based on input, backoff on amount of time to dial a peer again
+    pub fn has_dialed_recently(&self, backoff_duration: Duration) -> bool {
+        if let Ok(last_disconnect_duration) = self.last_dial_time.elapsed() {
```

Comment:
nit: `last_dial_duration` instead of `last_disconnect_duration` to be consistent and clear?

Reply:
Yeah, I renamed this, will fix